sshd-core/src/main/java/org/apache/sshd/common/util/Pair.java

```diff
@@ -45,6 +45,29 @@ public S getSecond() {
     }
 
     @Override
+    public int hashCode() {
+        return Objects.hashCode(first) * 31 + Objects.hashCode(second);
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (obj == null) {
+            return false;
+        }
+
+        if (obj == this) {
+            return true;
+        }
+
+        if (!(obj instanceof Pair)) {
+            return false;
+        }
+
+        Pair<?, ?> other = (Pair<?, ?>)obj;
+        return Objects.equals(first, other.first) && Objects.equals(second, other.second);
+    }
```

Comment:
Since _Pair_ is not _final_ (and neither are its _get_-ters) I prefer:
```java
return Objects.equals(getFirst(), other.getFirst()) && Objects.equals(getSecond(), other.getSecond());
```